### PR TITLE
Brodey | cublas + cudnn codegen

### PR DESCRIPTION
**Pull request by brodeynewman**

### CODE REVIEW

**Review by kevmo314 — APPROVED** [review]

File: codegen/codegen.py [hunk 1]
```diff
@@ -850,6 +915,12 @@ def main():
         for function, annotation, operations, disabled in functions_with_annotations:
             if function.name.format() in MANUAL_IMPLEMENTATIONS or disabled: continue
 
+            batched = False
+
+            # not a fan of this, but the batched functions are pretty standard with the flow below
+            if "Batched" in function.name.format():
```
Comment (kevmo314) [inline codegen/codegen.py:921]:
```py
batched = "Batched" in function.name.format()
```

Also can you add a comment describing what a batched function is? Would give some context to the conditional down below.

File: codegen/codegen.py [hunk 2]
```diff
@@ -487,7 +519,7 @@ def client_rpc_read(self, f):
 
 Operation = NullableOperation | ArrayOperation | NullTerminatedOperation | OpaqueTypeOperation | DereferenceOperation
 
-def parse_annotation(annotation: str, params: list[Parameter]) -> list[Operation]:
+def parse_annotation(annotation: str, params: list[Parameter]) -> list[Operation, bool]:
```
[the remark this answers is not in the document]
Comment (brodeynewman) [reply codegen/codegen.py:522]: describes if the function is disabled

[the remark this answers is not in the document]
Comment (brodeynewman) [reply codegen/codegen.py:522]: will comment

[the remark this answers is not in the document]
Comment (kevmo314) [reply codegen/codegen.py:522]: Isn't that `list[tuple[...]]`?

[the remark this answers is not in the document]
Comment (brodeynewman) [reply codegen/codegen.py:522]: hm, they seem to work synonymously. will add.

**Pull request merged**